### PR TITLE
chore: Logging AOT Support

### PR DESCRIPTION
**Pull request by hjgraca**
> Please provide the issue number

Issue number: #212 

### Changes

- Logging support for native AOT
- .NET 8 specific serialization paths
- New helper method to convert anonymous types to dictionary, use for source generation .NET 8
- Removed PowertoolsLambdaContext now uses ILambdaContext
- Removed dependency from IMethodAspectHandler, now using LoggingAspectFactory to boostrap Logging Aspect
- Added a new Source Generator - PowertoolsSourceGeneratorSerializer to be used in Lambda bootstrap 

### User experience

With the new PowertoolsSourceGeneratorSerializer, we can combine Powertools SerlializationContext with the customer Serialization context, this enables Powertools to serialize customer only types, this is useful for LogEvent feature.

image

## Checklist

Please leave checklist items unchecked if they do not apply to your change.

* [x] [Meets tenets criteria](https://docs.powertools.aws.dev/lambda/dotnet/tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-dotnet/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by hjgraca** [comment]
- [ ] documentation

File: libraries/src/AWS.Lambda.Powertools.Logging/Internal/LoggingAspect.cs [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    /// <summary>
+    ///     Captures the lambda context.
+    /// </summary>
+    /// <param name="eventArgs">
+    ///     The <see cref="T:AWS.Lambda.Powertools.Aspects.AspectEventArgs" /> instance containing the
+    ///     event data.
+    /// </param>
+    private void CaptureLambdaContext(AspectEventArgs eventArgs)
+    {
+        _clearLambdaContext = LoggingLambdaContext.Extract(eventArgs);
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Logging/Internal/LoggingAspect.cs:215]: Check where LambdaContext is used, if just here just use the Ilambda here

File: libraries/src/AWS.Lambda.Powertools.Logging/Serializers/PowertoolsLoggingSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    internal static string Serialize(object value)
+    {
+        var options = GetSerializerOptions();
+        return JsonSerializer.Serialize(value, options);
+    }
+#endif
+
+#if NET8_0_OR_GREATER
+
+    /// <summary>
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Logging/Serializers/PowertoolsLoggingSerializer.cs:74]: Merge methods so consumer dont need to have #if

**Review by amirkaws — APPROVED** [review]
LGTM

**Pull request merged**